src/pymor/operators/constructions.py

```diff
@@ -84,44 +84,56 @@ def evaluate_coefficients(self, mu):
 
     def apply(self, U, mu=None):
         coeffs = self.evaluate_coefficients(mu)
-        R = self.operators[0].apply(U, mu=mu)
-        R.scal(coeffs[0])
+        if coeffs[0]:
+            R = self.operators[0].apply(U, mu=mu)
+            R.scal(coeffs[0])
+        else:
+            R = self.range.zeros(len(U))
         for op, c in zip(self.operators[1:], coeffs[1:]):
-            R.axpy(c, op.apply(U, mu=mu))
+            if c:
+                R.axpy(c, op.apply(U, mu=mu))
         return R
 
     def apply2(self, V, U, mu=None):
         coeffs = self.evaluate_coefficients(mu)
-        matrices = [op.apply2(V, U, mu=mu) for op in self.operators]
+        try:
+            coeffs, matrices = zip(*[(c, self.operators[i].apply2(V, U, mu=mu))
+                                        for i, c in enumerate(coeffs) if c])
+        except ValueError:
+            return np.zeros((len(V), len(U)))
```

Comment:
I think that this should not be a try - except block since other errors of `apply2` .